core/code/ornaments.js

```diff
@@ -13,52 +13,120 @@
   OVERLAY_OPACITY: 0.6,
+  icon:[],
+  excludedOrnaments: [],
+  knownOrnaments: {},
 
   setup: function () {
     this._portals = {};
-    var layerGroup = L.layerGroup;
+    this.layerGroup = L.layerGroup;
     if (window.map.options.preferCanvas && L.Browser.canvas && !window.DISABLE_CANVASICONLAYER) {
-      layerGroup = L.canvasIconLayer;
+      this.layerGroup = L.canvasIconLayer;
       L.CanvasIconLayer.mergeOptions({ padding: L.Canvas.prototype.options.padding });
     }
-    this._layer = layerGroup();
-    this._beacons = layerGroup();
-    this._frackers = layerGroup();
-    window.layerChooser.addOverlay(this._layer, 'Ornaments');
-    window.layerChooser.addOverlay(this._beacons, 'Beacons');
-    window.layerChooser.addOverlay(this._frackers, 'Frackers');
+    this.load();
+
+    this.layers = {};
+    this.layers._layer = window.ornaments.layerGroup();
+    this.layers._excluded = window.ornaments.layerGroup(); // to keep excluded ornaments in an own layer
+
+    window.layerChooser.addOverlay(this.layers._layer, 'Ornaments');
+    window.layerChooser.addOverlay(this.layers._excluded, 'Excluded ornaments', {default: false});
+
+    $('<a>')
+      .html('Ornaments Opt')
+      .attr({
+        id: 'ornaments-toolbox-link',
+        title: 'Edit ornament exclusions',
+        accesskey: 'o'
+      })
+      .click(window.ornaments.ornamentsOpt)
+      .appendTo('#toolbox');
+  },
+  createLayer: function (layerID) {
+    window.ornaments.layers[layerID] = window.ornaments.layerGroup();
+    window.layerChooser.addOverlay(window.ornaments.layers[layerID], layerID);
   },
 
   addPortal: function (portal) {
     this.removePortal(portal);
-
     var ornaments = portal.options.data.ornaments;
     if (ornaments && ornaments.length) {
       this._portals[portal.options.guid] = ornaments.map(function (ornament) {
-        var layer = this._layer;
-        if (ornament.startsWith('pe')) {
-          layer = ornament === 'peFRACK'
-            ? this._frackers
-            : this._beacons;
-        }
+        var layer = this.layers._layer;
+        var opacity = this.OVERLAY_OPACITY;
         var size = this.OVERLAY_SIZE;
+        var anchor = [size / 2, size / 2];
+        var iconUrl = '//commondatastorage.googleapis.com/ingress.com/img/map_icons/marker_images/' + ornament + '.png';
+
+        if (!this.knownOrnaments[ornament]) {
+          this.knownOrnaments[ornament]=false;
+        }
+
+        if (typeof (window.ornaments.icon[ornament]) !== 'undefined') {
+          if (window.ornaments.icon[ornament].layer) {
+            if (window.ornaments.layers[window.ornaments.icon[ornament].layer] === undefined){
+              console.log ('Add missing layer: ',window.ornaments.icon[ornament].layer);
+              window.ornaments.createLayer(window.ornaments.icon[ornament].layer);
+            }
+            layer =  window.ornaments.layers[window.ornaments.icon[ornament].layer];
+          }
+          opacity = 1;
```

Comment:
I don't understand why you overwrite the opacity to 1 if the ornaments is redefined in a plugin.

Either the plugin changes `window.ornaments.OVERLAY_OPACITY` for all ornaments, or you can use a `.opacity` attribute that would be set in the `.icon[theOrnament]` data

Reply:
agreed, now optional parameter.